app/src/main/java/com/internship/move/utils/Utils.kt

```diff
@@ -36,4 +38,16 @@ fun interface ClickCallBack {
 
 fun logTag(tag: String, message: String = "") {
     println("[$tag] $message")
-}
\ No newline at end of file
+}
+
+fun checkMail(mailEditText: TextInputEditText): Boolean {
+    return Patterns.EMAIL_ADDRESS.matcher(mailEditText.text.toString()).matches() && mailEditText.text?.length!! > 7
+}
+
+fun checkUserOrPassword(userOrPasswordEditText: TextInputEditText): Boolean {
+    return userOrPasswordEditText.text?.length!! > 4
+}
+
+const val UNCHECKED = 0
+const val LOGGED = 1
+const val ERROR = -1
```

Comment:
Experiment with [Sealed Class](https://kotlinlang.org/docs/sealed-classes.html). 
This approach is not necessary wrong, but using sealed class would result in a much cleaner approach